synapseclient/producer_consumer_download_threads.py

```diff
@@ -0,0 +1,282 @@
+from queue import Queue
+from threading import Thread
+import typing
+import time
+from typing import Generator
+from re import split
+from datetime import datetime
+from math import ceil
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+from synapseclient.core.utils import printTransferProgress
+import json
+import os
+import requests
+
+# constants
+BUF_SIZE = 20
+MAX_RETRIES = 5
+MB = 2**20
+SYNAPSE_DEFAULT_DOWNLOAD_PART_SIZE = 8 * MB
+SYNAPSE_DEFAULT_FILE_ENDPOINT = "https://repo-prod.prod.sagebase.org/file/v1"
```

Comment:
Allow this to be configurable, base this off of client's current endpoint is.

Reply:
Is this different from `https://file-prod.prod.sagebase.org/file/v1`?  Because this is in the client:
```
synapseclient.client.PRODUCTION_ENDPOINTS
{'repoEndpoint': 'https://repo-prod.prod.sagebase.org/repo/v1',
 'authEndpoint': 'https://auth-prod.prod.sagebase.org/auth/v1',
 'fileHandleEndpoint': 'https://file-prod.prod.sagebase.org/file/v1',
 'portalEndpoint': 'https://www.synapse.org/'}
```

When you get a synapse connection you get 
```syn.fileHandleEndpoint
'https://file-prod.prod.sagebase.org/file/v1'```

Reply:
It can be different. Notice that the client also has staging endpoints. I hardcoded this endpoint for testing, but I have since changed it to use whatever file handle endpoint is being used by the client performing the download. It will be one of the file handle endpoints, not a repo one. Sorry for the confusion.

Reply:
No problem, I was just curious.